test/splitter.js

```diff
@@ -0,0 +1,160 @@
+const truffleAssert = require('truffle-assertions');
+const Splitter = artifacts.require("./Splitter.sol");
+
+contract('Splitter', async accounts => {
+
+    let gasCost = async receipt => {
+        const gasUsed = receipt.receipt.gasUsed;
+        const tx = await web3.eth.getTransaction(receipt.tx);
+        const gasPrice = tx.gasPrice;
+
+        return web3.utils.toBN(gasUsed * gasPrice);
+    };
+
+    let checkEventNotEmitted = async eventName => {
+        let result = await truffleAssert.createTransactionResult(splitter, splitter.transactionHash);
+
+        await truffleAssert.eventNotEmitted(
+            result,
+            eventName
+        );
+    }
+
+    const senderAddress = accounts[1];
+    const RecipientOneAddress = accounts[2];
+    const RecipientTwoAddress = accounts[3];
+    let splitter;
+
+    beforeEach("deploy and prepare", async function() {
+        splitter = await Splitter.new();
+    });
+
+    it('Sender sends 5 wei, 4 wei is split equally between recipients and 1 wei sent to the senders balance', async () => {
+
+        // Take snapshot of initial balances
+        let initSenderEthBalance = await web3.eth.getBalance(senderAddress);
+        initSenderEthBalance = web3.utils.toBN(initSenderEthBalance);
+
+        const initSenderContractBalance = await splitter.balances(senderAddress);
+        const initRecipientOneContractBalance = await splitter.balances(RecipientOneAddress);
+        const initRecipientTwoContractBalance = await splitter.balances(RecipientTwoAddress);
+
+        // Perform transactions
+        const receipt = await splitter.splitDeposit(RecipientOneAddress, RecipientTwoAddress, {from: senderAddress, value: 5});
+
+        const cost = await gasCost(receipt);
+        const transfer = web3.utils.toBN(5);
+        const splitAmount = web3.utils.toBN(2);
+        const remainder = web3.utils.toBN(1);
+
+        // Check sender's changed ETH balance
+        const expectedSenderEthBalance = initSenderEthBalance.sub(cost).sub(web3.utils.toBN(transfer));
+        let aliceEthBalance = await web3.eth.getBalance(senderAddress);
+        aliceEthBalance = web3.utils.toBN(aliceEthBalance);
+
+        assert.strictEqual(aliceEthBalance.toString(10), expectedSenderEthBalance.toString(10));
```

Comment:
What you are testing here is Web3 and Ethereum are correct. I.e. sending a transaction with value removes the value from my balance, but also that I paid the gas.
This is interesting but, as a test, it belongs to the Ethereum / Web3 battery of tests. Not here, where you test your contract.

Ok, your contract could mischievously immediately return some of the amount directly to the sender. But this is better tested by checking the contract balance instead of the sender's. By checking the contract balance, you check that the contract kept the amount instead of sending some to _any_ party, and not just the sender.